src/lib/store.js

```diff
@@ -499,124 +439,47 @@ export default class Store {
 		this.storeUpdate();
 	};
 
-	updateVendorList = vendorList => {
+	updateVendorList = (vendorList) => {
 		const {
-			pubVendorsList = {},
-			allowedVendorIds
-		} = this;
-
-		const {
-			created,
-			maxVendorId = 0
-		} = this.vendorConsentData;
-
-		if (vendorList) {
-			// Filter vendors in vendorList by allowedVendorIds
-			if (vendorList.vendors && allowedVendorIds.size) {
-				vendorList.vendors = vendorList.vendors.filter(({id}) => allowedVendorIds.has(id));
-			}
-
-			// If a pubVendorList is applied make the vendor list version = 0
-			const { publisherVendorsVersion } = pubVendorsList;
-			vendorList.vendorListVersion = publisherVendorsVersion ? 0 : vendorList.vendorListVersion;
-		}
+			created
+		} = this.persistedConsentData;
 
 		const {
-			vendors = [],
-			purposes = [],
+			vendors = {},
 		} = vendorList || {};
 
-		//collection of global and local vendor ids [global id, local id]
-		this.globalVendorIdsPresentOnList = new Map(vendors.filter(vendor => vendor.external_id).map(
-			vendor => [vendor.external_id, vendor.id]
-		));
+		const persistedMaxVendorId = this.persistedConsentData.vendorConsents && this.persistedConsentData.vendorConsents.maxId || 0;
+		this.tcModel.gvl = new GVL(vendorList);
+		this.vendorList = vendorList;
 
-		// If vendor consent data has never been persisted set default selected status
+		// If vendor and publisher consent data has never been persisted set default selected status
 		if (!created) {
-			this.vendorConsentData.selectedPurposeIds = new Set(purposes.map(p => p.id));
-			this.vendorConsentData.selectedVendorIds = new Set(vendors.map(v => v.id));
+			const getIds = (object) => Object.keys(object)
+				.filter(key => object[key].id)
+				.map(key => object[key].id);
+
+			const purposesIds = getIds(vendorList.purposes);
+			const vendorsIds = getIds(vendorList.vendors);
+			const specialFeatureIds = getIds(vendorList.specialFeatures);
+
+			this.tcModel.purposeConsents.set(purposesIds);
+			this.tcModel.purposeLegitimateInterests.set(purposesIds);
+			this.tcModel.vendorConsents.set(vendorsIds);
+			this.selectAllVendorLegitimateInterests(true, false);
+			this.tcModel.specialFeatureOptins.set(specialFeatureIds);
+			this.selectAllPublisherPurposes(true, false);
+			this.selectAllPublisherLegitimateInterests(true, false);
 		}
-
 		// If vendor consent data has already been persisted set default selected status only for new vendors
 		else {
-			vendors.forEach(v => {
-				if (v.id > this.persistedVendorConsentData.maxVendorId) {
-					this.vendorConsentData.selectedVendorIds.add(v.id);
+			Object.values(vendors).forEach(v => {
+				if (v.id > persistedMaxVendorId) {
+					this.tcModel.vendorConsents.set(v.id);
+					this.selectVendorLegitimateInterests(v.id, true);
```

Comment:
We notify store update listeners as many times as times as number of vendors with `legIntPurposes` here

Reply:
fixed